typing_extensions/src_py3/typing_extensions.py

```diff
@@ -2058,11 +2058,23 @@ class Annotated(metaclass=AnnotatedMeta):
 
 # Python 3.8 has get_origin() and get_args() but those implementations aren't
 # Annotated-aware, so we can't use those, only Python 3.9 versions will do.
-if sys.version_info[:2] >= (3, 9):
+# Similarly, Python 3.9's implementation doesn't support ParamSpecArgs and
```

Comment:
Another off topic comment: Something I just realised: if ParamSpecArgs/Kwargs had inherited from ``_GenericAlias`` in the original implementation, we wouldn't need this. Do you think it's worth it to do that in CPython or not worth the extra complexity/feel it's too hacky and not really what ``_GenericAlias`` actually represents?

Reply:
That's definitely an option. I feel like it's too different though. A `ParamSpec` isn't the type while the origin of a GenericAlias is, and ParamSpecArgs/Kwargs don't really have args that are comparable to those of a GenericAlias.

Reply:
Hmm yeah. Since ``ParamSpec`` isn't a Generic to begin with, ``GenericAlias`` definitely feels wrong.